traits/tests/test_trait_list_dict.py

```diff
@@ -115,3 +115,29 @@ def test_pickle_whole():
     loads(dumps(a))
     b = B(dict=dict(a=a))
     loads(dumps(b))
+
+
+def test_trait_set_object_operations():
```

Comment:
It would be nice to have tests for the normal case too (`a.myset |= another_set`, `a.myset | another_set`, etc.), but I think that's out of the scope of this PR.